File: csharp/triangle/Triangle.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class Triangle
{
    private static bool isValid(double[] ary)
    {
        Array.Sort(ary);
        return ary.All(c => c > 0) && ary[0] + ary[1] >= ary[2];
    }

    public static bool IsScalene(double side1, double side2, double side3)
    {
        var ary = new[] { side1, side2, side3 };
        var set = new HashSet<double>(ary);

        return isValid(ary) && set.Count == 3;
    }


    public static bool IsIsosceles(double side1, double side2, double side3)
    {
        var ary = new[] { side1, side2, side3 };
        var set = new HashSet<double>(ary);

        return isValid(ary) && set.Count <= 2;
    }

    public static bool IsEquilateral(double side1, double side2, double side3)
    {
        var ary = new[] { side1, side2, side3 };
        var set = new HashSet<double>(ary);

        return isValid(ary) && set.Count == 1;
    }
}

File: vbnet/triangle/Triangle.vb
Imports System
Public Module Triangle
    Public Function IsScalene(ByVal side1 As Double, ByVal side2 As Double, ByVal side3 As Double) As Boolean
        Return IsTriangle(side1, side2, side3) AndAlso new HashSet(Of Double)({side1, side2, side3}).Count = 3
    End Function
    Public Function IsIsosceles(ByVal side1 As Double, ByVal side2 As Double, ByVal side3 As Double) As Boolean
        Return IsTriangle(side1, side2, side3) AndAlso new HashSet(Of Double)({side1, side2, side3}).Count <= 2
    End Function
    Public Function IsEquilateral(ByVal side1 As Double, ByVal side2 As Double, ByVal side3 As Double) As Boolean
        Return IsTriangle(side1, side2, side3) AndAlso new HashSet(Of Double)({side1, side2, side3}).Count = 1
    End Function
    Private Function IsTriangle(ByVal side1 As Double, ByVal side2 As Double, ByVal side3 As Double) As Boolean
        Dim lst = new List(Of Double)({side1, side2, side3})
        lst.Sort()
        Return lst.All(Function(n) n > 0) AndAlso lst(0) + lst(1) >= lst(2)
    End Function
End Module
